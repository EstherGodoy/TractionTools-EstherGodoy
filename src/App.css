body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

button {
  padding: 0 15px;
}

/* HEADER */
header {
  background: #000;
  height: 60px;
  display: flex;
  align-items: center;
  color: #fff;
  padding:0 15px;
}

header h1 {
  font-weight: lighter;
  font-size: 22px;
}

header button {
  align-items: flex-end;
  margin-left: auto;
  height: 40px;
  border: 2px solid #fff;
  color: #fff;
  border-radius: none;
  background: transparent;
  font-size: 16px;
  width: 100px;
  background: #1DB954;
  cursor: pointer;
}

/* CATS */
.cats {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.cats li {
  width: calc(25% - 10px);
  padding: 5px;
}

.cats figure {
  position: relative;
  width: 100%;
  height: 250px;
  background-color: #eee;
  background-size: cover;
  background-position: center center;
  margin: 0;
}

.cats figure button {
  position: absolute;
  right: 10px;
  bottom: 10px;
  height: 40px;
  border: 0;
  color: #fff;
  border-radius: none;
  background: transparent;
  font-size: 16px;
  background: rgba(226, 169, 190, 1);
  cursor: pointer;
  display: none;
}

.cats figure:hover button {
  display: block;
}

.pagination {
  text-align: center;
  margin: 15px 0 0 0;
}

.pagination button {
  color: #fff;
  border: 0;
  border-radius: none;
  background: transparent;
  font-size: 16px;
  background: rgba(176, 171, 202, 1);
  cursor: pointer;
  height: 40px;
  margin: 0 5px;
}

.pagination button.next {
  background: rgba(163, 214, 212, 1);
}
