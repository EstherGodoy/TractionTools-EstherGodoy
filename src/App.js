import React, { Component } from "react";
import { Provider } from "mobx-react";
import { BrowserRouter, Route } from "react-router-dom";
import CatsStore from "./store";
import { decorate, observable, action } from "mobx";
import Home from "./components/views/page-home";
import "./styles/app.scss";

decorate(CatsStore, {
  catCount: observable,
  myFamily: observable,
  addToFamily: action,
  getFamily: action,
  removeFromFamily: action,
});

const catsStore = new CatsStore();

class App extends Component {
  render() {
    return (
      <Provider store={catsStore}>
        <BrowserRouter>
          <div className="container">
            <Route
              exact
              path="/"
              render={() => <Home history={this.props.history} />}
            />
          </div>
        </BrowserRouter>
      </Provider>
    );
  }
}

export default (App);
