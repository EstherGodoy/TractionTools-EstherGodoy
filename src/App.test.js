import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
import renderer from 'react-test-renderer';

//render test
it('application renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

//snapshot test
test("app snapshot test", (() => {
  const component = renderer.create(
    <App/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}))
