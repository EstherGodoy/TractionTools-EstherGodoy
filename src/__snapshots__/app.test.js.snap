// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot test 1`] = `
<div
  className="container"
>
  <header>
    <h1>
      If It Fits I Sits
    </h1>
    <button
      onClick={[Function]}
    >
      My Cat Fam 
    </button>
  </header>
  <section>
    <ul
      className="cats"
    />
    <nav
      className="pagination"
    >
      <button
        className="prev"
        disabled={true}
        onClick={[Function]}
      >
        Previous Page
      </button>
      <button
        className="next"
        disabled={true}
        onClick={[Function]}
      >
        Next Page
      </button>
    </nav>
  </section>
</div>
`;
